### PR TITLE
Remove IRunScript & IScriptHostControl and prepare IScriptOptionsProvider

### PR DESCRIPTION
**Pull request by mstv**
Supersedes #11321

## Proposed changes

- Remove IRunScript
- Turn empty `IScriptHostControl` into `IScriptOptionsProvider` with `string GetValue(string option)`

## Screenshots 

N/A

## Test methodology 

- existing tests

## Merge strategy

I agree that the maintainer squash merge this PR (if the commit message is clear).

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

File: GitUI/ScriptsEngine/IScriptOptionsProvider.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿namespace GitUI.ScriptsEngine;
+
+public interface IScriptOptionsProvider
+{
+    /// <summary>
+    /// Retrieves the information for placeholders in script arguments.
+    /// </summary>
+    /// <param name="option">The option identifier which is to be replaced.</param>
+    /// <returns>The value to be used for the script argument.</returns>
+    string GetValue(string option);
```
Comment (mstv) [inline GitUI/ScriptsEngine/IScriptOptionsProvider.cs:10]: Rather let the ScriptOptionsParser handle unsupported options:
```suggestion
    bool TryGetValue(string option, out string value);
```

File: GitUI/CommandsDialogs/UserScriptContextMenuExtensions.cs [hunk 2]
```diff
@@ -16,11 +16,13 @@ public static class UserScriptContextMenuExtensions
         /// <param name="contextMenu">The context menu to add user scripts to.</param>
         /// <param name="hostMenuItem">The menu item user scripts not marked as <see cref="ScriptInfo.AddToRevisionGridContextMenu"/> are added to.</param>
         /// <param name="scriptInvoker">The handler that handles user script invocation.</param>
-        public static void AddUserScripts(this ContextMenuStrip contextMenu, ToolStripMenuItem hostMenuItem, Action<int> scriptInvoker, IServiceProvider serviceProvider)
+        /// <param name="serviceProvider">The DI service provider.</param>
+        public static void AddUserScripts(this ContextMenuStrip contextMenu, ToolStripMenuItem hostMenuItem, Func<int, bool> scriptInvoker, IServiceProvider serviceProvider)
```
Comment (mstv) [inline GitUI/CommandsDialogs/UserScriptContextMenuExtensions.cs:20]: Just changed the type of `scriptInvoker` in order to directly be compatible with `GitModuleControl.ExecuteCommand`.

**Review by RussKie — APPROVED** [review]
nice

**Pull request merged**